packages/components/Tabs/src/Tabs.tsx

```diff
@@ -98,17 +98,23 @@ export const Tabs = compose<TabsType>({
 
     // Populate the tabsItemKeys array
     if (children) {
+      const enabledKeys = []
       // eslint-disable-next-line @typescript-eslint/ban-ts-comment
       // @ts-ignore - TODO, fix typing error
+      // Generates array of keys and enabled keys
       renderData.state.context.tabsItemKeys = React.Children.map(children, (child: React.ReactChild) => {
         if (React.isValidElement(child)) {
-          // Sets default selected tabItem
-          if (renderData.state.context.selectedKey == null && !child.props.disabled) {
-            renderData.state.context.selectedKey = child.props.itemKey;
+          if (!child.props.disabled) {
+            enabledKeys.push(child.props.itemKey);
           }
           return child.props.itemKey;
         }
       });
+
+      // Sets default selected tabsItem
```

Comment:
nit: might be worth adding a comment that default selected item is the first item if the given selected item is hidden.